app/internal/window/window.go

```diff
@@ -14,11 +14,61 @@ import (
 	"gioui.org/unit"
 )
 
+// Option configures a Window.
+type Option func(opts *Options)
+
+type optionCallback func()
+
+type (
+	OptSize struct {
+		optionCallback
+		Width, Height unit.Value
+	}
+	OptMinSize struct {
+		optionCallback
+		MinWidth, MinHeight unit.Value
+	}
+	OptMaxSize struct {
+		optionCallback
+		MaxWidth, MaxHeight unit.Value
+	}
+	OptTitle struct {
+		optionCallback
+		Title string
+	}
+)
+
 type Options struct {
-	Width, Height       unit.Value
-	MinWidth, MinHeight unit.Value
-	MaxWidth, MaxHeight unit.Value
-	Title               string
+	OptSize
+	OptMinSize
+	OptMaxSize
+	OptTitle
+}
+
+func (opts *Options) ApplyOption(option ...Option) {
+	old := *opts
+	for _, o := range option {
+		o(opts)
+	}
+
+	switch {
+	case old.Title != opts.Title:
+		if f := old.OptTitle.optionCallback; f != nil {
+			f()
+		}
+	case old.Width != opts.Width || old.Height != opts.Height:
+		if f := old.OptSize.optionCallback; f != nil {
+			f()
+		}
+	case old.MinWidth != opts.MinWidth || old.MinHeight != opts.MinHeight:
+		if f := old.OptMinSize.optionCallback; f != nil {
+			f()
+		}
+	case old.MaxWidth != opts.MaxWidth || old.MaxHeight != opts.MaxHeight:
+		if f := old.OptMaxSize.optionCallback; f != nil {
+			f()
+		}
+	}
```

Comment:
That doesn't seems right. But we need to identify what was changed, otherwise it will call all functions (such as Size) even when it's not intended. _Must be a better alternative for that._

Reply:
Let's drop the notion that an Option should be a function. I suggest

```
type Option struct {
    Title *string
    MaxWidth *int
    ...
}
```

So it's easy to tell which fields have changed. Option is not a pretty type, but it's only used internally.

Reply:
Switching to `struct` with pointer is better. But, why not create new interfaces? 

-----

Instead of change the `window.Window`, it's possible to add another interface:
```
type WindowBoundsSize interface {
SetMaxSize(x, y unit.Value)
SetMinSize(x, y unit.Value)
}
```

Then, create some `app.SetMaxSize(w window.Window, x, y unit.Value)`:

```
func SetMaxSize(w window.Window, x, y unit.Value) {
if ww, ok := w.(WindowBoundsSize); ok {
ww.SetMaxSize(x,y)
}
}
```

-----

In that case some platforms that doesn't support Resize/MaxSize/MinSize/Size (such as Android, JS, iOS) will never implement `WindowMaxSize`. I think it solves the problem of the forever growing `window.Window` interface. For instance (#10 ) adds alot of "no-op" functions just to implement the new window.Window. In that case, a new interface (such as `type WindowNavigationColor interface{}` and `type WindowStatusColor interface{}`) could be created.

Reply:
I believe my suggestion is simpler than interfaces, because you don't need new methods and the type-casting. Rather, backend use the settings field if they support it.

Another reason to use the Options struct is to support atomic settings. For example, setting dimensions and maxheight/width in one call. This is important to avoid flickering updates to the native window and for performance if the setting is expensive.